Userland/Shell/PosixParser.cpp

```diff
@@ -1437,15 +1458,20 @@ ErrorOr<RefPtr<AST::Node>> Parser::parse_for_clause()
         Optional<AST::Position> {});
 }
 
-RefPtr<AST::Node> Parser::parse_word_list()
+RefPtr<AST::Node> Parser::parse_word_list(bool allow_newlines)
 {
     Vector<NonnullRefPtr<AST::Node>> nodes;
 
     auto start_position = peek().position.value_or(empty_position());
 
+    while (allow_newlines && peek().type == Token::Type::Newline)
+        skip();
```

Comment:
This (and the other loop) reads a little weird since allow_newlines is not a changing condition of the loop. Maybe use an if instead, since the loop really only cares about the lookahead token type?

Reply:
it's more code for the exact same semantics, but sure